packages/flutter_tools/test/general.shard/android/android_device_start_test.dart

```diff
@@ -0,0 +1,274 @@
+          BuildInfo.release,
+        ),
+        platformArgs: <String, dynamic>{},
+      );
+
+      expect(launchResult.started, true);
+      expect(processManager.hasRemainingExpectations, false);
+    }, overrides: <Type, Generator>{
+      AndroidSdk: () => androidSdk,
+      FileSystem: () => fileSystem,
+      ProcessManager: () => processManager,
+    });
+  }
+
+  testUsingContext('AndroidDevice.startApp does not allow release builds on x86', () async {
+    const String deviceId = '1234';
+    final AndroidDevice device = AndroidDevice(deviceId, modelID: 'TestModel');
+    final File apkFile = fileSystem.file('app.apk')..createSync();
+    final AndroidApk apk = AndroidApk(
+      id: 'FlutterApp',
+      file: apkFile,
+      launchActivity: 'FLutterActivity',
+      versionCode: 1,
+    );
+
+    processManager.addCommand(kAdbVersionCommand);
+    processManager.addCommand(kStartServer);
+
+    // This configures the target platform of the device.
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'getprop'],
+      stdout: '[ro.product.cpu.abi]: [x86]'
+    ));
+
+    final LaunchResult launchResult = await device.startApp(
+      apk,
+      prebuiltApplication: true,
+      debuggingOptions: DebuggingOptions.disabled(
+        BuildInfo.release,
+      ),
+      platformArgs: <String, dynamic>{},
+    );
+
+    expect(launchResult.started, false);
+    expect(processManager.hasRemainingExpectations, false);
+  }, overrides: <Type, Generator>{
+    AndroidSdk: () => androidSdk,
+    FileSystem: () => fileSystem,
+    ProcessManager: () => processManager,
+  });
+
+  testUsingContext('AndroidDevice.startApp forwards all supported debugging options', () async {
+    const String deviceId = '1234';
+    final AndroidDevice device = AndroidDevice(deviceId, modelID: 'TestModel');
+    final File apkFile = fileSystem.file('app.apk')..createSync();
+    final AndroidApk apk = AndroidApk(
+      id: 'FlutterApp',
+      file: apkFile,
+      launchActivity: 'FLutterActivity',
+      versionCode: 1,
+    );
+
+    // These commands are required to install and start the app
+    processManager.addCommand(kAdbVersionCommand);
+    processManager.addCommand(kStartServer);
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'getprop']
+    ));
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'am', 'force-stop', 'FlutterApp'],
+    ));
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'pm', 'list', 'packages', 'FlutterApp'],
+    ));
+    processManager.addCommand(kAdbVersionCommand);
+    processManager.addCommand(kStartServer);
+    // TODO(jonahwilliams): investigate why this doesn't work.
+    // This doesn't work with the current Android log reader implementation.
+    processManager.addCommand(const FakeCommand(
```

Comment:
I can't get the observatory discovery to work for this case. Seems like the logs are getting dropped, even if I force the process to not exit immediately.